### PR TITLE
DS1307 RTC driver + example

### PR DESCRIPTION
**Pull request by UncleRus**
Driver for the most common RTC DS1307.

### CODE REVIEW

**Comment by ourairquality** [comment]
There is already a ds3231 driver, and I understood these were largely compatible, so is a separate driver really necessary?

**Comment by UncleRus** [comment]
Yes, they looks similar, but:
- registers map is same only at lower addresses (0x00 - 0x06)
- DS1307 has no alarms
- DS1307 has no temperature sensor
- Status and control registers are completely different
and so on.
I think specialized drivers for DS1307 and DS3231 is better.

**Comment by sheinz** [comment]
Thanks @UncleRus for the driver!

This looks really good to me.

Even though generic driver that supports both ds1307 and ds3231 seems like a reasonable approach. Most likely in a real project only one kind of chip will be used and a simple clean driver that supports only the hardware you have seems like a good thing. Nevertheless I don't think theres only one right way. For now this driver looks good as it is.

**Comment by UncleRus** [comment]
Please review

File: extras/ds1307/ds1307.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    DS1307_8192HZ,  //!< 8192 Hz
+    DS1307_32768HZ  //!< 32768 Hz
+} ds1307_squarewave_freq_t;
+
+/**
+ * \brief Init RTC
+ * \param scl_pin GPIO pin for I2C SCL
+ * \param sda_pin GPIO pin for I2C SDA
+ */
+void ds1307_init(uint8_t scl_pin, uint8_t sda_pin);
```
Comment (sheinz) [inline extras/ds1307/ds1307.h]: I think it was @ourairquality who pointed out that initialising I2C driver inside another driver might be inconvenient when you have several drivers in the system that use the same I2C.

**Comment by sheinz** [comment]
I've looked through the code and apart from a small comment about I2C initialisation everything looks good to me.

**Comment by sheinz** [comment]
I've just checked it on my hardware. The example worked from the first try :)

@UncleRus Could you squash your commits for this PR, and I will merge it.

**Comment by UncleRus** [comment]
Done

**Pull request merged**